### DISCUSSION TITLE
Node server throws error "Cannot find Module NanoID"

### DISCUSSION THREAD

**Discussion by TheOnePossum** [requarks/wiki#7361] [Error / Bug Report]
So i am trying to install Wiki.js on a Mac (M1/Sonoma 14.6.1). I've installed Node.js (like three times, just to be sure and not because I forgot I already installed it) and downloaded wiki.js from git as zip, unpacking it into /opt/homebrew/var/wikijs (It's right next to my www folder, cause I misunderstood some things in the instructions.)

Now upon opening the wiki.js folder in terminal and entering "node server" to install wiki.js, I get the following error:

node:internal/modules/cjs/loader:1228
  throw err;
  ^

Error: Cannot find module 'nanoid'
Require stack:
- /opt/homebrew/var/wikijs/server/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/opt/homebrew/var/wikijs/server/index.js:7:20)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Module._load (node:internal/modules/cjs/loader:1104:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/opt/homebrew/var/wikijs/server/index.js' ]
}

Node.js v20.17.0

What can I do here ?

Update:
NPM package manager allows to install nanoid using 'npm install module nanoid'
But that then throws:
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: wiki@2.0.0
npm error Found: graphql@15.3.0
npm error node_modules/graphql
npm error   graphql@"15.3.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer graphql@"^0.10.5 || ^0.11.3 || ^0.12.0 || ^0.13.0 || ^14.0.0" from graphql-subscriptions@1.1.0
npm error node_modules/graphql-subscriptions
npm error   graphql-subscriptions@"1.1.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /Users/*myusername*/.npm/_logs/2024-09-22T14_01_42_104Z-eresolve-report.txt
npm error A complete log of this run can be found in: /Users/*myusername*/.npm/_logs/2024-09-22T14_01_42_104Z-debug-0.log

Edit: Weirdest part:
NanoID isn't MEANT to have dependencies!

**Comment by NGPixel** [comment]
Follow the [install instructions](https://docs.requarks.io/install/macos)...

**Reply by TheOnePossum** [reply]
Those didn’t work for me, the Mac throws an error when trying to unpack the Tar. Nothing specific, something like “Cannot unpack tar” or something that way. Tried it four times.

**Comment by AnonymousWalker** [comment]
Same issue on Windows. Installation instruction didn't work.
